### PR TITLE
Feautre/add testing dependencies

### PR DESCRIPTION
**Pull request by akrabdev**
# Description

A unit testing setup using jest and react-testing-library that includes necessary configuration for Next.js.
Guided by:  [Next.js offical docs](https://nextjs.org/docs/testing#jest-and-react-testing-library)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Performed a simple test on the current Home component.

# Demo

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by samydevv — DISMISSED** [review]
Lgtm

**Review by raslan — DISMISSED** [review]

File: __tests__/index.test.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import Home from "../pages/index";
+
+describe("Home", () => {
+  it("renders a heading", () => {
```
Comment (raslan) [inline __tests__/index.test.tsx:10]:
```suggestion
  it("renders a Daisy button", () => {
```

**Review by raslan — CHANGES_REQUESTED** [review]
Please change this

**Review by raslan — APPROVED** [review]

**Review by Amrawy98 — APPROVED** [review]
good job

**Pull request merged**